get_s3_test.go

```diff
@@ -170,82 +169,77 @@ func TestS3Getter_ClientMode_collision(t *testing.T) {
 	}
 }
 
-func TestS3Getter_Minio_Url(t *testing.T) {
-	g := new(S3Getter)
-	forced, src := getForcedGetter("s3::http://127.0.0.1:9000/test-bucket/hello.txt?aws_access_key_id=TESTID&aws_access_key_secret=TestSecret&region=us-east-2&version=1")
-	u, err := url.Parse(src)
-
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	if forced != "s3" {
-		t.Fatalf("expected forced protocol to be s3")
-	}
-
-	region, bucket, path, version, creds, err := g.parseUrl(u)
-
-	if err != nil {
-		t.Fatalf("err: %s", err)
-	}
-
-	if region != "us-east-2" {
-		t.Fatal("expected us-east-2")
-	}
-
-	if bucket != "test-bucket" {
-		t.Fatal("expected test-bucket")
-	}
-
-	if path != "hello.txt" {
-		t.Fatal("expected hello.txt")
-	}
-
-	if version != "1" {
-		t.Fatal("expected to be 1")
-	}
-
-	if creds == nil {
-		t.Fatal("expected to not be nil")
-	}
+type parseTest struct {
+	url     string
+	region  string
+	bucket  string
+	path    string
+	version string
 }
 
-func TestS3Getter_S3_Url(t *testing.T) {
-	g := new(S3Getter)
-	forced, src := getForcedGetter("s3::https://s3-eu-west-1.amazonaws.com/bucket/foo/bar.baz?version=1234")
-	u, err := url.Parse(src)
-
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	if forced != "s3" {
-		t.Fatalf("expected forced protocol to be s3")
-	}
-
-	region, bucket, path, version, creds, err := g.parseUrl(u)
-
-	if err != nil {
-		t.Fatalf("err: %s", err)
-	}
-
-	if region != "eu-west-1" {
-		t.Fatal("expected eu-west-1")
-	}
-
-	if bucket != "bucket" {
-		t.Fatal("expected bucket")
-	}
-
-	if path != "foo/bar.baz" {
-		t.Fatal("expected foo/bar.baz")
-	}
-
-	if version != "1234" {
-		t.Fatal("expected to be 1234")
-	}
+var s3tests = []parseTest{
+	{
+		url:     "s3::https://s3-eu-west-1.amazonaws.com/bucket/foo/bar.baz?version=1234",
+		region:  "eu-west-1",
+		bucket:  "bucket",
+		path:    "foo/bar.baz",
+		version: "1234",
+	},
+	{
+		url:     "s3::http://127.0.0.1:9000/test-bucket/hello.txt?aws_access_key_id=TESTID&aws_access_key_secret=TestSecret&region=us-east-2&version=1",
+		region:  "us-east-2",
+		bucket:  "test-bucket",
+		path:    "hello.txt",
+		version: "1",
+	},
+	{
+		url:     "s3::http://127.0.0.1:9000/test-bucket/hello.txt?aws_access_key_id=TESTID&aws_access_key_secret=TestSecret&version=1",
+		region:  "us-east-1",
+		bucket:  "test-bucket",
+		path:    "hello.txt",
+		version: "1",
+	},
+	{
+		url:     "s3::http://127.0.0.1:9000/test-bucket/hello.txt?aws_access_key_id=TESTID&aws_access_key_secret=TestSecret",
+		region:  "us-east-1",
+		bucket:  "test-bucket",
+		path:    "hello.txt",
+		version: "",
+	},
+}
 
-	if creds != nil {
-		t.Fatal("expected to not be nil")
+func TestS3Getter_Url(t *testing.T) {
+	for i, pt := range s3tests {
```

Comment:
I would just inline here:

```
var s3tests = []struct{
    url     string
  ...
  } {
      {
         // test 1
      },
      ...
  }
  for i, pt := range s3tests {
  }
   ````

And then you can use the new table driven test using t.Run(..)

Better example here: https://blog.golang.org/subtests